src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -30,7 +30,8 @@ internal static bool IsNoOpSupported(RestoreRequest request)
         }
 
         /// <summary>
-        /// The cache file path is $(MSBuildProjectExtensionsPath)\$(project).nuget.cache
+        /// The cache file path will be $(RestoreOutputPath)\$(project).nuget.cache
```

Comment:
We prefer not to use RestoreOutputPath, so I'd revert this. 
RestoreOutputPath was added in the past, but we've moved away from that because RestoreOutputPath would lead to the *build* not importing the `nuget.g.*`. 

We even added a warning for that: https://github.com/NuGet/NuGet.Client/commit/deab84db30cbd8cc738cd46ad694aebdbb43a1cf
```suggestion
        /// The cache file path will be $(RestoreOutputPath)\$(project).nuget.cache
```

Reply:
Please correct me if I'm wrong. The `RestoreOutputPath` is used to define where the restore cache files should go and `MSBuildProjectExtensionsPath` is where we find the props/targets. Right?

Reply:
There are supposed to be the same. 

NuGet was defaulting to BaseIntermediateOutputPath, MSBuild was using MSBuildProjectExtensionsPath. These led to silent issues. 
At the same time, NuGet defined RestoreOutputPath and related which compounded the problem caused by 2 components using different props. 

The *only* right answer for end consumers is to use `MSBuildProjectExtensionsPath`.